internal/topology/variables/cluster_variable_validation.go

```diff
@@ -104,10 +104,9 @@ func ValidateClusterVariable(clusterVariable *clusterv1.ClusterVariable, cluster
 	}
 
 	// Convert schema to Kubernetes APIExtensions Schema.
-	apiExtensionsSchema, err := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema)
-	if err != nil {
-		return field.ErrorList{field.InternalError(fldPath,
-			fmt.Errorf("failed to convert schema definition for variable %q; ClusterClass should be checked: %v", clusterVariable.Name, err))} // TODO: consider if to add ClusterClass name
+	apiExtensionsSchema, allErrs := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema, fldPath)
+	if len(allErrs) > 0 {
+		return allErrs
 	}
```

Comment:
```suggestion
	apiExtensionsSchema, allErrs := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema, field.NewPath("schema"))
	if len(allErrs) > 0 {
		return field.ErrorList{field.InternalError(fldPath,
			fmt.Errorf("failed to convert schema definition for variable %q; ClusterClass should be checked: %v", clusterClassVariable.Name, allErrs))} // TODO: consider if to add ClusterClass name
	}
```

Similar issue & solution